### PR TITLE
Add support for preventDefault when using onError method

### PR DESCRIPTION
**Pull request by nochkin**
This is my proposed PR to support `preventDefault()` when using `onError` method as described in this [feature request](https://github.com/react-native-webview/react-native-webview/issues/1705).

### CODE REVIEW

File: docs/Reference.md [hunk 1]
```diff
@@ -265,6 +265,8 @@ Example:
 />
 ```
 
+The `syntheticEvent` can be stopped doing its default action by calling `syntheticEvent.preventDefault()`.
```
Comment (jamonholmgren) [inline docs/Reference.md:268]: I'd probably put this at the end of this section (just before the ### `onLoad` line).

**Review by jamonholmgren — CHANGES_REQUESTED** [review]
Looks straightforward enough! Just one minor tweak to the docs and should be ready to go.

File: docs/Reference.md (hunk 1 above)
Comment (nochkin) [reply docs/Reference.md]: Updated it according to your suggestion.

**Comment by nochkin** [comment]
I updated the documentation only and I see some tests are failing (one of them is failing due to timeout downloading packages).
Is there a way to re-run all tests without making any changes to PR?

**Comment by nochkin** [comment]
@jamonholmgren is there something else I need to do to make it merged? I updated the doc as suggested.

**Comment by matgardon** [comment]
+1 would be interested in this also, current behavior doesn't match my needs.

**Comment by nochkin** [comment]
@matgardon,
Yes, the PR was kinda approved at first and then I was asked to make a minor doc update which I did, and then it hangs somewhere in the space. I hope someone on a way to Mars will see it and pick it up. What are the chances?

**Comment by nochkin** [comment]
@jamonholmgren please merge.

**Review by Titozzz — APPROVED** [review]

**Comment by Titozzz** [comment]
I was on my way to mars...

This has been fixed by https://github.com/react-native-webview/react-native-webview/commit/279d6ae98c449b3c2d9ab2cc567f99bcc6e298b6.

Sorry for not coming back to you earlier. Thanks for fighting hard !

**Comment by nochkin** [comment]
@Titozzz 
Thanks for the follow up, but I see the fix you mentioned just removes some warnings in the console.
My proposed change actually adds a new behavior where you could handle errors in your own code instead of letting the RN-Webview to shows its own error for a user.

**Comment by Titozzz** [comment]
You are right. I thought the logs annoyed you, not the state. What's your usecase ? I'm worried this change might be an issue since it affects the internal state of the WebView

If we are being honest I'd rather remove all this state management as it doesn't make sense to deal with it inside the WebView but that's quite a breaking change

**Comment by nochkin** [comment]
The default behavior stays the same. If you want to handle your own errors, then you do `preventDefault`.
Unless someone did `preventDefault` for no reason (it won't do anything in the current version), it would not affect the way it's being used now.

You do 'preventDefault` in the `onError` callback only if you want this new behavior. Otherwise, it's default.

A particular use case is when you want to handle errors in your own app instead of letting RN-Webview spit out a generic error to a user. For example, you want to have a fallback procedure or come up with a message suggesting on how this can be resolved on the user's side.

Again, the existing apps would not see it unless there is `preventDefault` which triggers the new behavior.

Hopefully, that will make sense and not confuse you. I know Mars is pretty far :-) Let me know.

**Comment by Titozzz** [comment]
No no I understood you're changes but they are a small patch to a bigger issue. Why does WebView even handles it in a first place 😂. I'll merge when I get home

**Pull request merged**